Macrophages/Macrophages_multiome.py

```diff
@@ -0,0 +1,746 @@
+            heatmap_fig.update_layout(
+                    width=1400,
+                    height=400,
+                    margin=dict(l=100, r=100, t=100, b=100),
+                    xaxis_title=" ",
+                    yaxis_title=" ",
+                    title_x=0.5,  # Center the title
+                    font={'size': 16, 'family': 'Arial'},
+                )
+            heatmap_fig.update_xaxes(showticklabels=False)
+            return heatmap_fig
+    # Return an empty figure if no data is available
+    return {}
+
+def process_data_for_table(molecular_name):
+    """
+    Helper function to process the DataFrame for the DataTable.
+    """
+    filtered_df = select_df_of_same_cluster(df_mean_multi_clustered, molecular_name)
+    if not filtered_df.empty:
+        # Reset the index to convert it into a column
+        reset_df = filtered_df.reset_index()
+        # Optionally, rename the index column for clarity
+        reset_df.rename(columns={'index': 'Index'}, inplace=True)
+        # Convert DataFrame to dictionary format for DataTable
+        data = reset_df.to_dict('records')
+        # Define table columns including the new index column
+        columns = [{"name": col, "id": col} for col in reset_df.columns]
+        return columns, data, ""
+    else:
+        return [], [], f"Selected {molecular_name} is not significantly dysregulated in all treatments!!!"
+
+
+def extract_unique_gene_names(gene_list):
+    """
+    Extract unique gene names from a list of input strings, ensuring the output list is the same length as the input list.
+
+    Parameters:
+    gene_list (list): List of strings containing multiple gene entries.
+
+    Returns:
+    list: List of gene names extracted from the input strings, maintaining the same length as the input list.
+    """
+    gene_names = set()
+    result_genes = []
+    pattern = re.compile(r'\|([A-Za-z0-9]+)_[A-Za-z]+\b')
+
+    for entry in gene_list:
+        # Find all matches of the pattern in the current string
+        matches = pattern.findall(entry)
+        selected_gene = None
+        if matches:
+            for gene in matches:
+                if gene not in gene_names:
+                    gene_names.add(gene)
+                    selected_gene = gene
+                    break
+            # If no new gene is found, select the first gene in the list
+            if not selected_gene:
+                selected_gene = matches[0]
+        else:
+            # If no matches are found, keep the original input string
+            selected_gene = entry
+        result_genes.append(selected_gene)
+
+    return result_genes
+
+# get gene to pathway reflection
+dfdf = df_mean_multi_clustered.copy()
+dfdf.index = extract_unique_gene_names(df_mean_multi_clustered.index)
+gene_value_dict = dfdf.iloc[:,:4].to_dict(orient='index')
+meta_gene = dfdf.iloc[-73:,:]
+# list(meta_gene[meta_gene['kmeans']==0].index)
+
+import networkx as nx

```

Comment:
I think Lex is making this comment, but you may want to break these up into separate files (multiple import statements). If you need help learning how to access functions/classes from other files, feel free to reach out to Lex or myself.